python/HistogramTools.py

```diff
@@ -2,6 +2,36 @@
 
 import ROOT
 import re
+from types import MethodType
+
+class TFileWrapper(ROOT.TFile):
+    """Act as a wrapper around TFile, to make the 'Get' operation faster by keeping an internal dictionary of the file's keys."""
+
+    def __init__(self, *args, **kwargs):
+        ROOT.TFile.__init__(self, *args, **kwargs)
+        self.keys = { key.GetName(): key for key in self.GetListOfKeys() }
+
+    def Get(self, name):
+        key = self.keys.get(name, None)
+        if key:
+            return key.ReadObj()
+        else:
+            raise Exception("Could not read object {} from file.".format(name))
+
+    @staticmethod
+    def GetStatic(self, name):
+        key = self.keys.get(name, None)
+        if key:
+            return key.ReadObj()
+        else:
+            raise Exception("Could not read object {} from file.".format(name))
+    
+    @staticmethod
+    def Open(*args):
+        _file = ROOT.TFile.Open(*args)
+        _file.keys = { key.GetName(): key for key in _file.GetListOfKeys() }
+        _file.Get = MethodType(TFileWrapper.GetStatic, _file)
```

Comment:
I tried but python complained because `Get` is an instance method... Maybe there's another way?